homeassistant/components/melcloud/sensor.py

```diff
@@ -8,83 +13,86 @@
     STATE_CLASS_MEASUREMENT,
     SensorEntity,
 )
-from homeassistant.const import (
-    ATTR_DEVICE_CLASS,
-    ATTR_ICON,
-    ENERGY_KILO_WATT_HOUR,
-    TEMP_CELSIUS,
-)
+from homeassistant.const import ENERGY_KILO_WATT_HOUR, TEMP_CELSIUS
 from homeassistant.util import dt as dt_util
 
 from . import MelCloudDevice
 from .const import DOMAIN
 
-ATTR_MEASUREMENT_NAME = "measurement_name"
-ATTR_UNIT = "unit"
-ATTR_VALUE_FN = "value_fn"
-ATTR_ENABLED_FN = "enabled"
-
-ATA_SENSORS = {
-    "room_temperature": {
-        ATTR_MEASUREMENT_NAME: "Room Temperature",
-        ATTR_ICON: "mdi:thermometer",
-        ATTR_UNIT: TEMP_CELSIUS,
-        ATTR_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
-        ATTR_VALUE_FN: lambda x: x.device.room_temperature,
-        ATTR_ENABLED_FN: lambda x: True,
-    },
-    "energy": {
-        ATTR_MEASUREMENT_NAME: "Energy",
-        ATTR_ICON: "mdi:factory",
-        ATTR_UNIT: ENERGY_KILO_WATT_HOUR,
-        ATTR_DEVICE_CLASS: DEVICE_CLASS_ENERGY,
-        ATTR_VALUE_FN: lambda x: x.device.total_energy_consumed,
-        ATTR_ENABLED_FN: lambda x: x.device.has_energy_consumed_meter,
-    },
+
+@dataclass
+class SensorMetadata:
+    """Metadata for an individual sensor."""
+
+    measurement_name: str
+    icon: str
+    unit: str
+    device_class: str
+    value_fn: Callable[[Any], float]
+    enabled: Callable[[Any], bool]
+
+
+ATA_SENSORS: dict[str, SensorMetadata] = {
+    "room_temperature": SensorMetadata(
+        "Room Temperature",
+        "mdi:thermometer",
+        TEMP_CELSIUS,
+        DEVICE_CLASS_TEMPERATURE,
+        lambda x: x.device.room_temperature,
+        lambda x: True,
+    ),
```

Comment:
I would again suggest to add keywords. In this case for everything except `measurement_name`.